### PR TITLE
clientkit: add HTTP rest API

### PR DESCRIPTION
**Pull request by webispy**
add NuguHttpRest class to support HTTP request.

Signed-off-by: Inho Oh <webispy@gmail.com>

### CODE REVIEW

File: include/clientkit/nugu_http_rest.hh [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+#include <functional>
+#include <map>
+#include <string>
+
+#include <base/nugu_http.h>
+
+namespace NuguClientKit {
+
+/**
+ * @file http_rest.hh
```
Comment (kimhyungrok) [inline include/clientkit/nugu_http_rest.hh:29]: Is it correct [nugu_http_rest.hh]?

File: src/clientkit/nugu_http_rest.cc [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+NuguHttpResponse* NuguHttpRest::del(const std::string& path,
+    const NuguHttpHeader* additional_header)
+{
+    NuguHttpRequest* req;
+    NuguHttpResponse* resp;
+    NuguHttpHeader* header = make_header(common_header, additional_header);
+
+    req = nugu_http_delete_sync(host, path.c_str(), header);
+    if (req == NULL) {
+        nugu_error("nugu_http_delete_sync() failed");
```
Comment (kimhyungrok) [inline src/clientkit/nugu_http_rest.cc:247]: Is there need to free header?

File: include/clientkit/nugu_http_rest.hh (hunk 1 above)
Comment (webispy) [reply include/clientkit/nugu_http_rest.hh:29]: Thank you! I will update the file name.

File: src/clientkit/nugu_http_rest.cc (hunk 2 above)
Comment (webispy) [reply src/clientkit/nugu_http_rest.cc:247]: Thank you! I will add the `nugu_http_header_free()`

**Review by kimhyungrok — APPROVED** [review]

**Pull request merged**